tests/indexes/arithmetic/test_floordiv.py

```diff
@@ -0,0 +1,127 @@
+
+import numpy as np
+import pandas as pd
+import pytest
+from typing_extensions import (
+    Never,
+    assert_type,
+)
+
+from tests import (
+    TYPE_CHECKING_INVALID_USAGE,
+    check,
+)
+
+
+@pytest.fixture
+def left_i() -> pd.Index:
+    """Left operand"""
+    lo = pd.MultiIndex.from_arrays([[1, 2, 3]]).levels[0]
+    return check(assert_type(lo, pd.Index), pd.Index, np.integer)
+
+
+def test_floordiv_py_scalar(left_i: pd.Index) -> None:
+    """Test pd.Index[int] // Python native scalars"""
+    b, i, f, c = True, 1, 1.0, 1j
+    s, d = datetime(2025, 9, 27), timedelta(seconds=1)
+
+    check(assert_type(left_i // b, pd.Index), pd.Index, np.integer)
+    check(assert_type(left_i // i, pd.Index), pd.Index, np.integer)
+    check(assert_type(left_i // f, pd.Index), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _03 = left_i // c  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _04 = left_i // s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _05 = left_i // d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    check(assert_type(b // left_i, pd.Index), pd.Index, np.integer)
+    check(assert_type(i // left_i, pd.Index), pd.Index, np.integer)
+    check(assert_type(f // left_i, pd.Index), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _13 = c // left_i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _14 = s // left_i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+    check(assert_type(d // left_i, pd.TimedeltaIndex), pd.TimedeltaIndex, pd.Timedelta)
+
+
+def test_floordiv_py_sequence(left_i: pd.Index) -> None:
+    """Test pd.Index[int] // Python native sequences"""
+    b, i, f, c = [True, True, True], [2, 3, 5], [1.0, 2.0, 3.0], [1j, 1j, 4j]
+    s = [datetime(2025, 10, 27 + d) for d in range(3)]
+    d = [timedelta(seconds=s) for s in range(3)]
+
+    check(assert_type(left_i // b, pd.Index), pd.Index, np.integer)
+    check(assert_type(left_i // i, pd.Index), pd.Index, np.integer)
+    check(assert_type(left_i // f, pd.Index), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _03 = left_i // c  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _04 = left_i // s  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _05 = left_i // d  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+
+    check(assert_type(b // left_i, pd.Index), pd.Index, np.integer)
+    check(assert_type(i // left_i, pd.Index), pd.Index, np.integer)
+    check(assert_type(f // left_i, pd.Index), pd.Index, np.floating)
+    if TYPE_CHECKING_INVALID_USAGE:
+        _13 = c // left_i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _14 = s // left_i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
+        _15 = d // left_i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue]
```

Comment:
this works

Reply:
In this case, can we just return `Index` without specifying what is inside? i.e., `Index[Any]` or just `Index` ?